helix-term/src/ui/editor.rs

```diff
@@ -770,6 +770,73 @@ impl Component for EditorView {
 
                 EventResult::Consumed(callback)
             }
+            Event::Mouse(MouseEvent {
+                kind: MouseEventKind::Down(MouseButton::Left),
+                row,
+                column,
+                modifiers,
+                ..
+            }) => {
+                let editor = &mut cx.editor;
+                let tree = &mut editor.tree;
+                let mut id: Option<ViewId> = None;
+                let mut result = EventResult::Ignored;
+
+                for (view, view_id) in tree.views_with_keys_mut() {
```

Comment:
Rather than for loop which will iterate over everything, use .find` since views_mut() is already an iterator.

Reply:
Also rather than having a verify method and one that returns a position, you can use find_map

Reply:
You mean `pos_at_screen_coords` should not validate its input and thus panic? Well, with doc comment stating that should be absolutely fine.

Reply:
I meant https://github.com/helix-editor/helix/pull/509#discussion_r678282400 <- find_map already does the verification